src/liballoc/tests/btree/map.rs

```diff
@@ -554,6 +554,26 @@ fn test_clone() {
     }
 }
 
+#[test]
+fn test_clone_from() {
```

Comment:
This does test various sizes, with both source bigger than destination and destination bigger than source. Do you mean bigger maps being tested? I could extend rather than insert at each iteration. Either way, I'll move the `insert` statements in the loop to below the `clone_from` checks, which will add empty cases.